tests/sync/sync_manager.cpp

```diff
@@ -338,9 +339,41 @@ TEST_CASE("sync_manager: file actions", "[sync]") {
             REQUIRE_REALM_DOES_NOT_EXIST(realm_path_2);
             REQUIRE_REALM_DOES_NOT_EXIST(realm_path_3);
             // There should be recovery files.
-            CHECK(File::exists(util::file_path_by_appending_component(recovery_dir, recovery_1)));
-            CHECK(File::exists(util::file_path_by_appending_component(recovery_dir, recovery_2)));
-            CHECK(File::exists(util::file_path_by_appending_component(recovery_dir, recovery_3)));
+            CHECK(File::exists(recovery_1));
+            CHECK(File::exists(recovery_2));
+            CHECK(File::exists(recovery_3));
+        }
+
+        SECTION("should copy the Realm to the recovery_directory_path") {
+            SyncManager::shared().configure_file_system(base_path, SyncManager::MetadataMode::NoEncryption);
+            const std::string identity = "b241922032489d4836ecd0c82d0445f0";
+            const auto realm_base_path = file_manager.user_directory(identity) + "realmtasks";
+
+            REQUIRE(create_dummy_realm(realm_base_path));
+            REQUIRE_REALM_EXISTS(realm_base_path);
+
+            // Manually create a file action metadata entry to move the file to a destination
+            std::string recovery_path = util::reserve_unique_file_name(SyncManager::shared().recovery_directory_path(),
+                                                                       util::create_timestamped_template("recovered_realm"));
+            // Ensure recovery file doesn't exist yet
+            REQUIRE(!File::exists(recovery_path));
+
+            // Open the metadata separately, so we can investigate it ourselves.
+            SyncMetadataManager manager(file_manager.metadata_path(), false);
+            auto a = SyncFileActionMetadata(manager, SyncFileActionMetadata::Action::HandleRealmForClientReset, realm_base_path, "", "", recovery_path);
+
+            SyncManager::shared().reset_for_testing();
```

Comment:
`reset_for_testing()` only resets the sync manager's in-memory state. What the test is trying to do is to simulate the app starting up again and reloading the persisted metadata (following line). Should I rename `reset_for_testing()` to make its semantics clearer?